src/test/java/teammates/test/cases/webapi/CreateFeedbackResponseActionTest.java

```diff
@@ -23,13 +50,167 @@ protected String getRequestMethod() {
+        String[] notAnswerableForStudents = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn2InGracePeriodInCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+        };
+        verifyCannotAccess(notAnswerableForStudents);
+        ______TS("not answerable to instructors");
+        String[] notAnswerableForInstructors = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+        };
+        verifyCannotAccess(notAnswerableForInstructors);
+        ______TS("invalid HTTP parameters");
+        String[] invalidParams = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.FULL_DETAIL.toString(),
+        };
+        assertThrows(InvalidHttpParameterException.class, () -> getAction(invalidParams).checkAccessControl());
+        invalidParams[3] = Intent.INSTRUCTOR_RESULT.toString();
+        assertThrows(InvalidHttpParameterException.class, () -> getAction(invalidParams).checkAccessControl());
+
+        ______TS("invalid recipient for student");
+        loginAsStudent(student1InCourse1.getGoogleId());
+        String[] invalidRecipientForStudent = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+        };
+        assertThrows(UnauthorizedAccessException.class, () -> {
+            FeedbackResponseCreateRequest createRequest = getResponseRequest(student2InCourse1.getEmail());
+            getAction(createRequest, invalidRecipientForStudent).checkAccessControl();
+        });
+        ______TS("invalid recipient for instructor");
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+        String[] invalidRecipientForInstructor = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn2InGracePeriodInCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+        };
+        assertThrows(UnauthorizedAccessException.class, () -> {
+            FeedbackResponseCreateRequest createRequest = getResponseRequest(instructor1OfCourse2.getEmail());
+            getAction(createRequest, invalidRecipientForInstructor).checkAccessControl();
+        });
+
+    }
+
+    private FeedbackResponseCreateRequest getResponseRequest(String email) {
+        FeedbackResponseCreateRequest createRequest = new FeedbackResponseCreateRequest();
+        createRequest.setQuestionType(FeedbackQuestionType.TEXT);
+        createRequest.setRecipientIdentifier(email);
+        FeedbackResponseDetails responseDetails = new FeedbackTextResponseDetails("This is the text");
+        createRequest.setResponseDetails(responseDetails);
+        return createRequest;
     }
 
+    @Override
+    protected void prepareTestData() {
```

Comment:
I think it is better to override `beforeTestMethodSetup` and call `super.beforeTestMethodSetup()`.

In addition, it is better to put the setup method at the beginning.